src/core/public/chrome/ui/header/home_loader.tsx

```diff
@@ -0,0 +1,143 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import './home_loader.scss';
+
+import { i18n } from '@osd/i18n';
+import React from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { Observable } from 'rxjs';
+import Url from 'url';
+import { EuiHeaderSectionItemButton } from '@elastic/eui';
+import { ChromeNavLink } from '../..';
+import { ChromeBranding } from '../../chrome_service';
+import { LoadingIndicator } from '../loading_indicator';
+import { HomeIcon } from './home_icon';
+
+function findClosestAnchor(element: HTMLElement): HTMLAnchorElement | void {
+  let current = element;
+  while (current) {
+    if (current.tagName === 'A') {
+      return current as HTMLAnchorElement;
+    }
+
+    if (!current.parentElement || current.parentElement === document.body) {
+      return undefined;
+    }
+
+    current = current.parentElement;
+  }
+}
+
+function onClick(
+  event: React.MouseEvent<HTMLAnchorElement>,
+  forceNavigation: boolean,
+  navLinks: ChromeNavLink[],
+  navigateToApp: (appId: string) => void
+) {
+  const anchor = findClosestAnchor((event as any).nativeEvent.target);
+  if (!anchor) {
+    return;
+  }
+
+  const navLink = navLinks.find((item) => item.href === anchor.href);
+  if (navLink && navLink.disabled) {
+    event.preventDefault();
+    return;
+  }
+
+  if (event.isDefaultPrevented() || event.altKey || event.metaKey || event.ctrlKey) {
+    return;
+  }
+
+  if (forceNavigation) {
+    const toParsed = Url.parse(anchor.href);
+    const fromParsed = Url.parse(document.location.href);
+    const sameProto = toParsed.protocol === fromParsed.protocol;
+    const sameHost = toParsed.host === fromParsed.host;
+    const samePath = toParsed.path === fromParsed.path;
+
+    if (sameProto && sameHost && samePath) {
+      if (toParsed.hash) {
+        document.location.reload();
+      }
+
+      // event.preventDefault() keeps the browser from seeing the new url as an update
```

Comment:
+1 for such useful comments

Reply:
Not my code, but I agree 😆 . I actually have a feeling that none of this navigation handling code is still necessary, and is vestigial from when angular was handling the routing. But it was safest to just directly move the functionality as it currently exists for the scope of this PR.